instance/tests/test_instance_upgrade.py

```diff
@@ -0,0 +1,252 @@
+        ("Test", "Miss", "Miss", "Test"),  # replacement miss
+        ("ASD\nZXC", "ASD", "QWE", "QWE\nZXC"),  # multiline replacement - single hit
+        ("ASD\nASD", "ASD", "QWE", "QWE\nQWE"),  # multiline replacement - multiple hits
+        ("ASD\nASD", "ASD\n", "", "ASD"),  # remove line
+    )
+    def test_replace_extra_config(self, initial_settings, target, replacement, expected_settings):
+        """
+        Test greping configuration_extra_settings
+        """
+        instance_mock = self._make_instance_mock()
+        instance_mock.configuration_extra_settings = initial_settings
+        self.upgrader.replace_extra_config(instance_mock, target, replacement)
+        self.assertEqual(instance_mock.configuration_extra_settings, expected_settings)
+
+    def test_upgrade_instance(self):
+        """
+        Test upgrade_instances
+        """
+        instances_collection = [self._make_instance_mock() for _ in range(3)]
+
+        with patch.object(self.upgrader, 'get_instances_to_upgrade') as patched_get_instances, \
+                patch.object(self.upgrader, 'upgrade_instance') as patched_upgrade_instance:
+            patched_get_instances.return_value = instances_collection
+            self.upgrader.upgrade_instances()
+
+            for instance in instances_collection:
+                instance.save.assert_called_once_with()
+
+            expected_upgrade_instance_calls = [call(instance) for instance in instances_collection]
+            self.assertEqual(patched_upgrade_instance.mock_calls, expected_upgrade_instance_calls)
+
+            expected_spawn_calls = [
+                call(instance.ref.pk, mark_active_on_success=True, num_attempts=1)
+                for instance in instances_collection
+            ]
+            self.assertEqual(self.spawn_appserver_mock.mock_calls, expected_spawn_calls)
+
+
+class TestDogwoodToEucalyptus1(BaseInstanceUpgradeTest):
+    """
+    Test upgrade from dogwood to eucalyptus.1
+    """
+    def setUp(self):
+        """
+        Test set up method
+        """
+        super(TestDogwoodToEucalyptus1, self).setUp()
+        self.upgrader = DogwoodToEucalyptus1()
+
+    def test_upgrade_instance(self):
+        """
+        Test upgrading single instance
+        """
+        instance_mock = self._make_instance_mock()
+        revision = "opencraft-release/eucalyptus.1"
+        expected_platform_spec = self._make_git_spec("https://github.com/open-craft/edx-platform", revision)
+        expected_config_spec = self._make_git_spec("https://github.com/open-craft/configuration", revision)
+
+        # precondition check
+        self._assert_git_specs(instance_mock, expected_platform_spec, expected_config_spec, not_equal=True)
+
+        self.upgrader.upgrade_instance(instance_mock)
+        self._assert_git_specs(instance_mock, expected_platform_spec, expected_config_spec)
+        self.assertIn("COMMON_EUCALYPTUS_UPGRADE: true", instance_mock.configuration_extra_settings)
```

Comment:
@e-kolpakov It might be worth checking for newlines as well:

```python
self.assertIn("\nCOMMON_EUCALYPTUS_UPGRADE: true\n", instance_mock.configuration_extra_settings)
```